torch/csrc/distributed/c10d/reducer.cpp

```diff
@@ -729,6 +734,29 @@ void Reducer::finalize_backward() {
   local_used_maps_reduced_ = false;
 }
 
+void runGradCallbaclForVariable(
+    torch::autograd::Variable& variable,
+    GradCallback&& cb) {
+  auto context_ptr = rpc_context_.context_ptr.load();
```

Comment:
Why do we need this rpc_context_? Why can't we just lookup the thread local directly by calling something like ThreadLocalDistAutogradContext::getContext()?

Reply:
This is to avoid possible data race. I suppose multiple threads may be calling the post hooks referencing to the same reducer instance.